lib/interface_quda.cpp

```diff
@@ -3251,6 +3230,328 @@ void invertQuda(void *hp_x, void *hp_b, QudaInvertParam *param)
+    v_c[0] = cloverPrecise;
+    quda::split_field(*collected_clover, v_c, split_key); // Clover uses 4d even-odd preconditioning.
+    // Free the original clover fields all together.
+    freeCloverQuda();
+
+    // Set up the new re-distributed clover fields.
+    cloverPrecise = static_cast<quda::cudaCloverField *>(collected_clover);
+    QudaPrecision prec[] = {param->clover_cuda_prec_sloppy, param->clover_cuda_prec_precondition,
+                            param->clover_cuda_prec_refinement_sloppy, param->clover_cuda_prec_eigensolver};
+    loadSloppyCloverQuda(prec);
+  }
+
+  quda::GaugeField *collected_gauge = nullptr;
+  quda::GaugeField *collected_milc_fatlink_field = nullptr;
+  quda::GaugeField *collected_milc_longlink_field = nullptr;
+
+  if (!is_staggered) {
+    gf_param->create = QUDA_NULL_FIELD_CREATE;
+    collected_gauge = new quda::cpuGaugeField(*gf_param);
+    std::vector<quda::GaugeField *> v_g(1);
+    v_g[0] = in;
+    quda::split_field(*collected_gauge, v_g, split_key);
+  } else {
+    milc_fatlink_param->create = QUDA_NULL_FIELD_CREATE;
+    milc_longlink_param->create = QUDA_NULL_FIELD_CREATE;
+    collected_milc_fatlink_field = new quda::cpuGaugeField(*milc_fatlink_param);
+    collected_milc_longlink_field = new quda::cpuGaugeField(*milc_longlink_param);
+    std::vector<quda::GaugeField *> v_g(1);
+    v_g[0] = milc_fatlink_field;
+    quda::split_field(*collected_milc_fatlink_field, v_g, split_key);
+    v_g[0] = milc_longlink_field;
+    quda::split_field(*collected_milc_longlink_field, v_g, split_key);
+  }
+
+  profileInvertSplitGrid.TPSTOP(QUDA_PROFILE_INIT);
+  profileInvertSplitGrid.TPSTART(QUDA_PROFILE_PREAMBLE);
+
+  comm_barrier();
+
+  // Split input fermion field
+  quda::ColorSpinorParam cpu_cs_param_split(*_h_x[0]);
+  for (int d = 0; d < CommKey::n_dim; d++) { cpu_cs_param_split.x[d] *= split_key[d]; }
+  std::vector<quda::ColorSpinorField *> _collect_b(param->num_src_per_sub_partition, nullptr);
+  std::vector<quda::ColorSpinorField *> _collect_x(param->num_src_per_sub_partition, nullptr);
+  for (int n = 0; n < param->num_src_per_sub_partition; n++) {
+    _collect_b[n] = new quda::cpuColorSpinorField(cpu_cs_param_split);
+    _collect_x[n] = new quda::cpuColorSpinorField(cpu_cs_param_split);
+    auto first = _h_b.begin() + n * num_sub_partition;
+    auto last = _h_b.begin() + (n + 1) * num_sub_partition;
+    std::vector<ColorSpinorField *> _v_b(first, last);
+    split_field(*_collect_b[n], _v_b, split_key, pc_type);
+  }
+  comm_barrier();
+
+  push_communicator(split_key);
+  updateR();
+  comm_barrier();
+
+  // Load gauge field after pushing the split communicator so the comm buffers, etc are setup according to
+  // the split topology.
+  if (!is_staggered) {
+    loadGaugeQuda(collected_gauge->Gauge_p(), gauge_param);
+  } else {
+    loadFatLongGaugeQuda(param, gauge_param, collected_milc_fatlink_field->Gauge_p(), collected_milc_longlink_field->Gauge_p());
+  }
+
+  /** For debug
```

Comment:
I would suggest either deleting this commented out debug code or guarding it with either a C macro or by `getVerbosity() >= QUDA_DEBUG_VERBOSE`